packages/base/src/config/ThemeRoots.js

```diff
@@ -0,0 +1,70 @@
+import { getThemeRoots as getConfiguredThemeRoots } from "../InitialConfiguration.js";
+import createLinkInHead from "../util/createLinkInHead.js";
+import removeLinkFromHead from "../util/removeLinkFromHead.js";
+import applyTheme from "../theming/applyTheme.js";
+import { getTheme } from "./Theme.js";
+
+// {
+//     "custom_theme": "url"
+//     "custom_theme2": {
+//         "sap.ui.core": "url"
+//     }
+// }
+
+let themeRoots;
+
+const getThemeRoot = theme => {
+	return getThemeRoots()[theme];
+};
+
+const getThemeRoots = () => {
+	if (themeRoots === undefined) {
+		themeRoots = getConfiguredThemeRoots();
+	}
+
+	return themeRoots;
+};
+
+const setThemeRoot = (theme, themeRoot) => {
+	themeRoot += (themeRoot.slice(-1) === "/" ? "" : "/");
+
+	themeRoots[theme] = themeRoot;
+
+	attachStylesToHead();
+};
+
+const formatThemeLink = theme => {
+	return `${themeRoots[theme]}Base/baseLib/${theme}/css_variables.css`;
+};
+
+const createThemeLink = theme => {
```

Comment:
Passing a callback parameter is generally fine, but creating an `async` function is preferable, if appropriate. Please consider returning a promise here